test/get_outs/BUILD

```diff
@@ -0,0 +1,90 @@
+nonamedouts = genrule(
+    name = "nonamedouts",
+    outs = ["x"],
+    cmd = """
+        echo 'x' > "$OUTS"
+    """,
+)
+
+gr_getouts = genrule(
+    name = "genrule_getouts",
+    outs = {
+        "wibble": ["wibble_file1"],
+        "wobble": ["wobble_file1"],
+    },
+    cmd = """
+        echo 'wibblewibblewibble' > "$OUTS_WIBBLE"
+        echo 'wobblewobblewobble' > "$OUTS_WOBBLE"
+    """,
+)
+
+fg_getouts = filegroup(
+    name = "filegroup_getouts",
+    srcs = {
+        "wibble": [text_file(name = 'wibble_file2', content = 'wibblewibblewibble')],
+        "wobble": [text_file(name = 'wobble_file2', content = 'wobblewobblewobble')],
+    },
+)
+
+def assert_dict(l1, l2):
+    if l1 != l2:
+        fail(f"{l1} != {l2}")
+
+assert_dict({}, get_named_outs(nonamedouts))
+
+assert_dict({
+    "wibble": ["//test/get_outs:genrule_getouts|wibble"],
+    "wobble": ["//test/get_outs:genrule_getouts|wobble"],
+}, get_named_outs(gr_getouts))
+
+assert_dict({
+    "wibble": ["//test/get_outs:filegroup_getouts|wibble"],
+    "wobble": ["//test/get_outs:filegroup_getouts|wobble"],
```

Comment:
Sorry! I was thinking more that the dict would return a mapping of the named output tag to the outs that constitute it i.e. 
```
{
    "wibble": ["wibble_file1"],
    "wobble": ["wobble_file1"]
}
```

Does that make sense? Apologies for the back and forth.

Reply:
No apologies necessary, It makes more sense with what get_outs is doing indeed.
I think it will make the result less useful to use with an another rule, but that's fine, one can still regenerate the `AnnotatedOutputLabel` using the keys.

Maybe something better for the usage I'm describing would be `get_target_outs` that would return the same thing as right now ?

What do you think?

Reply:
I made the change you requested